tests/system/src/test/java/test/robot/javafx/stage/StageFocusTest.java

```diff
@@ -0,0 +1,141 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.stage;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.paint.Color;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+import test.util.Util;
+import test.robot.testharness.VisualTestBase;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class StageFocusTest extends VisualTestBase {
+
+    static CountDownLatch startupLatch;
+    static CountDownLatch eventReceivedLatch;
+
+    static final int STAGE_SIZE = 200;
+
+    static final int STAGE_X = 100;
+    static final int STAGE_Y = 100;
+
+    static final int TIMEOUT = 2000; // ms
+    static final double TOLERANCE = 0.07;
+
+    static final Color SCENE_COLOR = Color.LIGHTGREEN;
+
+    private Stage theStage = null;
+
+    @BeforeAll
+    public static void setupOnce() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        eventReceivedLatch = new CountDownLatch(1);
+    }
+
+    /**
+     * Checks whether Stage is actually shown when calling show()
+     *
+     * Meant as a "canary" test of sorts to ensure other tests relying on
+     * Stage being actually shown and on foreground work fine.
+     */
+    @Test
+    public void testStageHasFocusAfterShow() throws InterruptedException {
+        Util.runAndWait(() -> {
+            theStage = getStage(false);
+
+            Group root = new Group();
+            Scene scene = new Scene(root, STAGE_SIZE, STAGE_SIZE);
+            scene.setFill(SCENE_COLOR);
+            scene.setOnKeyPressed(e -> {
+                if (e.getCode() == KeyCode.A) {
+                    eventReceivedLatch.countDown();
+                }
+            });
+
+            theStage.setScene(scene);
+            theStage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> {
+                Platform.runLater(() -> {
+                    theStage.setX(STAGE_X);
+                    theStage.setY(STAGE_Y);
```

Comment:
setting Stage (x,y) coordinates in runLater makes the window jump.
it's probably better to move LL98-99 before runLater, to L96

with the robot disabled, the test fails with expected
```
org.opentest4j.AssertionFailedError: Event received latch timed out! Stage most probably did not have focus after showing. Some tests might fail because of this. If that is the case, try re-running the tests with '--no-daemon' flag in Gradle. ==> expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
```